### DISCUSSION TITLE
Why the region not indexed be included into grains when we do grain construction?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#1848] [Ask Anything]
Hi, everyone,
when I do grain reconstruction, the notindexed region is assigned to the surrounding grains, resulting in a larger grain size.
How can I leave them unindexed?

my code is :
 thetac_recon = 10; %degree
    [grain_recon,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree, 10*degree]);

![image](https://github.com/mtex-toolbox/mtex/assets/69452222/a092038e-8c02-4030-8ea3-2f70d3e29729)
![image](https://github.com/mtex-toolbox/mtex/assets/69452222/05ea218d-be60-4fee-bf5a-a121a19f2b20)

**Comment by kilir** [comment]
Hi,
compute grains using the entire dataset, not just the 'indexed' phases.
E.g. ` [grain_recon,ebsd.grainId] = calcGrains(ebsd);`
Cheers,
Rüdiger

**Reply by Brian-Liu-rx** [reply]
Thanks Rüdiger
